src/components/alert/alert.e2e.ts

```diff
@@ -269,25 +270,31 @@ describe("calcite-alert", () => {
     });
   });
 
-  it("should emit proper 'calciteAlertOpen' and 'calciteAlertClose' events on animation end", async () => {
+  it("should emit component status for transition-chained events: 'calciteAlertBeforeOpen', 'calciteAlertOpen', 'calciteAlertBeforeClose', 'calciteAlertClose'", async () => {
     const page = await newE2EPage();
-    await page.setContent(`<calcite-alert>
-    ${alertContent}
-    </calcite-alert>`);
+    await page.setContent(html`<calcite-alert> ${alertContent} </calcite-alert>`);
 
     const element = await page.find("calcite-alert");
-    const container = await page.find(`calcite-alert >>> .container`);
+    const container = await page.find(`calcite-alert >>> .${CSS.container}`);
+
+    expect(await container.isVisible()).toBe(false);
 
+    const beforeOpenEvent = page.waitForEvent("calciteAlertBeforeOpen");
     const openEvent = page.waitForEvent("calciteAlertOpen");
+
     element.setAttribute("active", "");
     await page.waitForChanges();
+    await beforeOpenEvent;
     await openEvent;
 
     expect(await container.isVisible()).toBe(true);
 
+    const beforeCloseEvent = page.waitForEvent("calciteAlertBeforeClose");
     const closeEvent = page.waitForEvent("calciteAlertClose");
+
     element.removeAttribute("active");
     await page.waitForChanges();
+    await beforeCloseEvent;
     await closeEvent;
```

Comment:
Could we add some event spy assertions in this test as well?  It would be cool to spy on all 4 component state events (beforeOpen, open, beforeClosing, closed) and make sure that they each only fire once per interaction.